fix: LIIKUNTA-422 Fixed map search layout to be more consistant with …

HeikkM opened the pull request:
LIIKUNTA-422:
Layout changes. Matched the search element of the map seach page to match other pages. 

![closed](https://user-images.githubusercontent.com/125858958/227114350-39f39c16-8529-48e5-9bba-ef6401f53008.jpg)
![Open](https://user-images.githubusercontent.com/125858958/227114381-48f0ca57-c941-4059-a265-6150ad9816a4.png)

LIIKUNTA 429
Removed breadcrumb property from page component. 
Removed breadcrumbs from events, hobbies, sports src/pages/[...slug].txs. 

![old](https://user-images.githubusercontent.com/125858958/227133075-2efcb549-d5ef-4e7e-a0fc-bc2deee00065.png)
![fix](https://user-images.githubusercontent.com/125858958/227133091-1536a478-2887-4eb7-bbd7-8f8a5a14c865.png)

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela commented on apps/events-helsinki/src/pages/pages/[...slug].tsx:
```diff
@@ -102,7 +99,6 @@ type ResultProps =
   | {
       initialApolloState: NormalizedCacheObject;
       page: PageQuery['page'];
-      breadcrumbs: Breadcrumb[];
```
I would have kept this field in the type.

HeikkM replied:
Made a new commit that returns this.

nikomakela commented on apps/sports-helsinki/src/domain/search/searchHeader/SearchHeader.tsx:
```diff
@@ -25,77 +32,94 @@ type Props = {
-                theme="black"
-                iconLeft={<IconMenuHamburger aria-hidden="true" />}
-                onClick={switchShowMode}
-              >
-                {t('common:mapSearch.showAsList')}
-              </Button>
-              <div className={styles.countAndTags}>
-                <Text variant="h3">
-                  {count} {t('common:mapSearch.searchResultsCountLabel')}
-                </Text>
-              </div>
-              <Button
-                variant="secondary"
-                theme="black"
-                iconLeft={<IconSearch aria-hidden="true" />}
-                onClick={() => {
-                  setCollapsed(false);
+            )}
+            {!collapsed && (
+              <PageSection>
+                <div className={styles.horizontalDivider}></div>
+              </PageSection>
+            )}
```
An old code refactoring advice: I think it would make sense to move this part to a new (private) function / component or even split it in 2 new functions. It would make the code cleaner.

nikomakela commented on apps/sports-helsinki/src/domain/search/searchHeader/searchHeader.module.scss:
```diff
@@ -2,7 +2,7 @@
 
 .searchHeader {
-  --background-color: #{$color-engel-medium-light};
+  --background-color: var(--background-color);
```
isn't this just reassigning an empty variable again?

nikomakela commented on apps/sports-helsinki/src/domain/search/venueSearch/VenueMapSearch.tsx:
```diff
@@ -9,7 +9,7 @@ const SimpleVenueMapSearch = () => (
     searchUtilities={null}
     korosBottom={false}
-    showTitle={false}
+    showTitle={true}
```
```suggestion
    showTitle
```
the `{true}` is not needed and actually our linter should advice to use it without. 

_I don't know why our linters are softer nowadays (I think it's been so after the commit that had ErrorBoundary and lot's of refactoring)._

nikomakela commented on apps/sports-helsinki/src/domain/search/venueSearch/VenueSearch.tsx:
```diff
@@ -7,14 +7,14 @@ import {
   useCommonTranslation,
   MultiSelectDropdown,
   getUrlParamAsArray,
+  // eslint-disable-next-line import/no-unresolved
```
This doesn't look right. This should not be needed. `events-helsinki-components` is very much needed and it must be resolvable.

nikomakela commented on apps/sports-helsinki/src/pages/search/map.tsx:
```diff
@@ -1,17 +1,22 @@
   Navigation,
   MatomoWrapper,
   getLanguageOrDefault,
+  // eslint-disable-next-line import/no-unresolved
```
This doesn't look right. This should not be needed. events-helsinki-components is very much needed and it must be resolvable.

nikomakela commented on apps/sports-helsinki/src/pages/search/map.tsx:
```diff
@@ -27,6 +32,7 @@ import type { Connection, SearchResult } from '../../types';
 
 // https://stackoverflow.com/a/64634759
 const MapView = dynamic(
+  // eslint-disable-next-line import/no-unresolved
```
Should not be needed - must be resolvable.

nikomakela reviewed (changes requested):
It sure does look better now, even though I think that this transition from white "cloud" to a map looks a bit rough. But that is a design decision!
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/389204/227253616-ece7ec6a-537c-40d1-8d90-f1273f45cf75.png">
<img width="426" alt="image" src="https://user-images.githubusercontent.com/389204/227253701-5e23d372-02ce-4dd4-b22a-cd3b34967342.png">


Some changes requested:
- There are several linter ignores added that should never be used.
- Remove the newly added personal env-file

There are also some minor and trivial advices.

HeikkM committed:
fix: removing unused breadcrumbs

LIIKUNTA-429. Removed breadcrumbs component from page.

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr275.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr275.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela commented on apps/sports-helsinki/.env.local.example:
```diff
@@ -1,32 +0,0 @@
-NEXT_PUBLIC_FEDERATION_ROUTER_ENDPOINT="https://events-graphql-federation-sports.test.hel.ninja/"
```
This is the example file for local private environment config. This should not be deleted!

- `.env.local.example` is there for developers. Its needed. Note the dot-prefix.
- `.env.local` is in use, but it's in gitignore, because it's a private local file. Note the dot-prefix.


There should not be any other env-related files at the moment.

nikomakela reviewed (changes requested):
The env-config example should not be removed. The rest is OK

nikomakela commented:
I tested this in my Google Pixel 5. The buttons are hardly visible. Some background is needed to them in mobile view.

![Screenshot_20230328-103220.png](https://user-images.githubusercontent.com/389204/228162208-f3dc875d-2eee-455e-92b4-b0d3537e3e41.png)

![Screenshot_20230328-103200.png](https://user-images.githubusercontent.com/389204/228162209-36b0742e-ff27-441e-836d-7ee6e34848de.png)

HeikkM commented:
For smaller sized, removed footer(was added earlier) and changed button background color. Any Changes regarind the mobile will be taken to new task. There was some suggestions from Ville.

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela commented on apps/sports-helsinki/src/pages/search/map.tsx:
```diff
@@ -1,17 +1,22 @@
+  // eslint-disable-next-line import/no-unresolved
 } from 'events-helsinki-components';
 import { LoadingSpinner } from 'hds-react';
+import { t } from 'i18next';
```
A new review was requested, but this is still here, so I want to raise this up again. We want to use our own instance of the i18next, so we should always avoid using the `tFunction` from i18next. The one I suggested is a type aware and configured instance.

nikomakela commented on apps/sports-helsinki/src/domain/search/searchHeader/searchHeader.module.scss:
```diff
@@ -50,31 +49,35 @@
     bottom: 0;
     left: 0;
     right: 0;
+    padding: 0 0 $spacing-layout-xs 0;
```
Is there a real design reasoning why this xs-padding is set to the bottom? Because of that, there is a transparent line under the button-row:

<img width="432" alt="image" src="https://user-images.githubusercontent.com/389204/228857065-969a83e1-5c6f-4d59-aa27-4c7a71b9cb4d.png">


```suggestion
```

nikomakela commented on apps/sports-helsinki/src/domain/search/searchHeader/searchHeader.module.scss:
```diff
@@ -83,33 +86,29 @@
 }
 
+.paddingClosed {
+  padding: $spacing-layout-xs 0 $spacing-layout-xs 0 !important;
```
When the search-section is closed, the cap is filled with white (instead of transparent like when it's opened), but it's still unnecessary in my opinion:

<img width="462" alt="image" src="https://user-images.githubusercontent.com/389204/228857679-5ae2f418-46cb-4be0-a098-3320517a990c.png">

```suggestion
```

nikomakela reviewed (changes requested):
Even better!  Now the button bar is visible in my Pixel 5 and the UI is more usable.

I would still like to ask about these bottom paddings in mobile view button bar. I suggested some changes and here is the render result with those changes active:

<img width="437" alt="image" src="https://user-images.githubusercontent.com/389204/228858845-0518602e-4927-40ac-a437-86020f54a0da.png">

<img width="440" alt="image" src="https://user-images.githubusercontent.com/389204/228858911-3d75a9bb-cda6-4530-ab21-1fd4f8bc7f4e.png">

The i18next import that I mentioned earlier is still not fixed and I think it's a must have.

nikomakela reviewed (changes requested):
Why is this? This shouldn't have been done
`apps/sports-helsinki/.env.local.example` → `apps/sports-helsinki/.env.example.local`

The only app related env-file that we want in our repository is this `.env.local.example` and it should not be renamed.

HeikkM committed:
chore: map layout to be similar with the list view

LIIKUNTA-422

nikomakela reviewed (approved):
Good! The code changes are now good enough, but there is still a conflict with the main branch. _That of course needs to be addressed before the merge. Also the commits should be squashed all to one (or 2), because they now are like internal fixes to each others inside one scope feature scope.:_

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr275.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr275.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr275.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr275.dev.hel.ninja :laughing::tada::tada::tada:
